diff -up phonon-mpv-master/src/mediaobject.cpp.2~ phonon-mpv-master/src/mediaobject.cpp
--- phonon-mpv-master/src/mediaobject.cpp.2~	2023-11-20 21:09:33.389915673 +0100
+++ phonon-mpv-master/src/mediaobject.cpp	2023-11-20 21:10:24.116463254 +0100
@@ -25,6 +25,7 @@
 #include <QDir>
 #include <QStringBuilder>
 #include <QUrl>
+#include <QList>
 
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
 #include <QLatin1StringView>
@@ -44,6 +45,8 @@ static const int ABOUT_TO_FINISH_TIME =
 
 using namespace Phonon::MPV;
 
+static QList<void*> recentlyDestroyed;
+
 MediaObject::MediaObject(QObject* parent)
     : QObject(parent)
     , m_nextSource(MediaSource(QUrl()))
@@ -77,9 +80,14 @@ MediaObject::MediaObject(QObject* parent
 }
 
 MediaObject::~MediaObject() {
+    recentlyDestroyed.append(this);
 }
 
 void MediaObject::event_cb(void *opaque) {
+    if(recentlyDestroyed.contains(opaque)) {
+        recentlyDestroyed.removeAll(opaque);
+        return;
+    }
     MediaObject* that = reinterpret_cast<MediaObject*>(opaque);
     QMetaObject::invokeMethod(
                     that,
